proc_macros/src/rule_tests.rs

```diff
@@ -3,20 +3,100 @@ use quote::{format_ident, quote, ToTokens};
 use syn::{
     braced, bracketed,
     parse::{Parse, ParseStream},
-    parse_macro_input, Expr, ExprArray, Ident, Token,
+    parse_macro_input, token, Expr, ExprArray, Ident, Token,
 };
 
+struct RuleOptions {
+    options: Expr,
```

Comment:
This I'd think might get fancier so that if a rule takes eg an "options-style object" you can write something in the `rule_tests!()` macro that looks like about what you'd expect to be able to write for that?

(specifically it's not going through a deserialization step like the options from the config file are so it might be awkward to try and have them literally write it as eg YAML/JSON syntax? But I haven't really wrapped my head around that)